<launch>

    <!-- Arguments -->
    <arg name="robot_name" /> <!--name will change with multiple robots-->
    <arg name="init_pose" /> <!--init pos will change for each robot-->
    <arg name="model" default="$(env TURTLEBOT3_MODEL)" /> <!--model type [burger, waffle, waffle_pi]-->

    <!-- Load the specific robot decription for tb3_0 -->
    <param name="robot_description"
        command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

    <!-- Start the robot state publisher -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher_0">
        <param name="publish_frequency" type="double" value="50.0" />
    </node>

    <!-- Spawn the model in gazebo -->
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf_0"
        args="$(arg init_pose) -urdf -model $(arg robot_name) -param robot_description" />

</launch>